### PR TITLE
Adding check to ensure 'container' get's added to guest_tech when appropriate

### PR DESCRIPTION
**Pull request by carlosmmatos**
##### SUMMARY
Enhancment/patch for #70832 to enhance reporting of guest_tech type 'container'. 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/facts/virtual/linux.py

### CODE REVIEW

**Comment by carlosmmatos** [comment]
@relrod let me know what you think.

**Comment by carlosmmatos** [comment]
Just checking in

File: lib/ansible/module_utils/facts/virtual/linux.py [hunk 1]
```diff
@@ -104,6 +104,11 @@ def get_virtual_facts(self):
                 virtual_facts['virtualization_role'] = 'guest'
                 found_virt = True
 
+        # ensure 'container' guest_tech is appropriately set
+        if 'container' not in guest_tech:
```
Comment (relrod) [inline lib/ansible/module_utils/facts/virtual/linux.py]: This shouldn't be necessary, since it's a set. It won't get added twice no matter what.

**Comment by relrod** [comment]
@carlosmmatos can you add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment)

File: lib/ansible/module_utils/facts/virtual/linux.py (hunk 1 above)
Comment (carlosmmatos) [reply lib/ansible/module_utils/facts/virtual/linux.py]: good point - let me update

**Comment by carlosmmatos** [comment]
@relrod will do

**Comment by carlosmmatos** [comment]
@relrod Am I missing something on the CI failure messages? Or is this some infra issue?

**Comment by carlosmmatos** [comment]
seems to always be an issue on the AWS side...I can try forcing it to run again

**Pull request merged**